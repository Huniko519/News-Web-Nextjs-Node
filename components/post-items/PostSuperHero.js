import React from 'react';
import PropTypes from 'prop-types';
import PostItemMedia from './PostItemMedia';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';

/**
 * Display a large image with the headline and exerpt
 * ╔══════════╤══════════════╗
 * ║ Headline │              ║
 * ║          │    Image     ║
 * ║ Excerpt  │              ║
 * ╚══════════╧══════════════╝
 * @param {post: {...postData}} props
 */
const PostSuperHero = (props) => {
  const { post } = props;
  const isSponsored = post.type === 'post_sponsored';

  return (
    <div className="inews__post inews__post-superhero" key={post.id} data-post-id={post.id}>
      <div className="inews__post-superhero__content">
        <PostItemTitle post={post} />
        <PostItemLabel post={post} />
        {isSponsored && (
          <span className="inews__post__sponsored-label">
            Promoted Content
          </span>
        )}
        <p
          dangerouslySetInnerHTML={{
            __html: post.excerpt.rendered,
          }}
        />
      </div>
      <div>
        {/* div is needed for positioning context for video icon to prevent edge cases with grid */}
        <PostItemMedia
          className="inews__post-superhero__media inews__post-superhero__media--1-1"
          imageSize="medium-1:1"
          width={1}
          height={1}
          layout="responsive"
          placeholder="/static/images/placeholder/placeholder-84x84.png"
          post={post}
        />
        <PostItemMedia
          className="inews__post-superhero__media inews__post-superhero__media--16-9"
          imageSize="medium-16:9"
          width={16}
          height={9}
          layout="responsive"
          placeholder="/static/images/placeholder/placeholder-640x360.png"
          post={post}
        />
      </div>
    </div>
  );
};

/* istanbul ignore next */
PostSuperHero.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default PostSuperHero;
