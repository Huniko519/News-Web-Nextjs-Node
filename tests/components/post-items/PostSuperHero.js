import React from 'react';
import PostSuperHero from '../../../components/post-items/PostSuperHero';

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postSuperHero = shallow(<PostSuperHero post={post} />);
  expect(postSuperHero).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postSuperHero = shallow(<PostSuperHero post={post} />);
  expect(postSuperHero.hasClass('inews__post')).toBe(true);
  expect(postSuperHero.hasClass('inews__post-superhero')).toBe(true);
});

it('should call the <PostItemMedia /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postSuperHero = shallow(<PostSuperHero post={post} />);
  const postItemMedia = postSuperHero.find('PostItemMedia');
  expect(postItemMedia.length).toBe(2);
  expect(postItemMedia.at(0).prop('className')).toBe('inews__post-superhero__media inews__post-superhero__media--1-1');
  expect(postItemMedia.at(0).prop('imageSize')).toBe('medium-1:1');
  expect(postItemMedia.at(0).prop('width')).toBe(1);
  expect(postItemMedia.at(0).prop('height')).toBe(1);
  expect(postItemMedia.at(0).prop('layout')).toBe('responsive');
  expect(postItemMedia.at(0).prop('post')).toMatchObject(post);

  expect(postItemMedia.at(1).prop('className')).toBe('inews__post-superhero__media inews__post-superhero__media--16-9');
  expect(postItemMedia.at(1).prop('imageSize')).toBe('medium-16:9');
  expect(postItemMedia.at(1).prop('width')).toBe(16);
  expect(postItemMedia.at(1).prop('height')).toBe(9);
  expect(postItemMedia.at(1).prop('layout')).toBe('responsive');
  expect(postItemMedia.at(1).prop('post')).toMatchObject(post);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postSuperHero = shallow(<PostSuperHero post={post} />);
  const postItemTitle = postSuperHero.find('PostItemTitle');
  expect(postItemTitle.length).toBe(1);
  expect(postItemTitle.prop('post')).toMatchObject(post);
});

it('should output a Sponsored label if post.type === post_sponsored', () => {
  const post = {
    id: 1,
    type: 'post_sponsored',
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postSuperHero = shallow(<PostSuperHero post={post} />);
  const label = postSuperHero.find('span.inews__post__sponsored-label');
  expect(label.length).toBe(1);
  expect(label.text()).toBe('Promoted Content');
});
