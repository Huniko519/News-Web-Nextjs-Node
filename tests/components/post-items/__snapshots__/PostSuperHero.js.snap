// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-superhero"
  data-post-id={1}
  key="1"
>
  <div
    className="inews__post-superhero__content"
  >
    <PostItemTitle
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
    />
    <PostItemLabel
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
    />
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "excerpt",
        }
      }
    />
  </div>
  <div>
    <PostItemMedia
      className="inews__post-superhero__media inews__post-superhero__media--1-1"
      height={1}
      imageSize="medium-1:1"
      layout="responsive"
      linkToAuthor={false}
      placeholder="/static/images/placeholder/placeholder-84x84.png"
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      postItemType=""
      showAuthor={false}
      width={1}
    />
    <PostItemMedia
      className="inews__post-superhero__media inews__post-superhero__media--16-9"
      height={9}
      imageSize="medium-16:9"
      layout="responsive"
      linkToAuthor={false}
      placeholder="/static/images/placeholder/placeholder-640x360.png"
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      postItemType=""
      showAuthor={false}
      width={16}
    />
  </div>
</div>
`;
